### PR TITLE
feat: [AG-108] integrate Hashflow

### PR DESCRIPTION
**Pull request by lehainam-dev**
- Add Hashflow integration.

### CODE REVIEW

File: pkg/source/hashflow/client/config.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
+package client
+
+import (
+	"github.com/KyberNetwork/kyberswap-dex-lib/pkg/util/duration"
+	"github.com/KyberNetwork/kyberswap-dex-lib/pkg/valueobject"
+)
+
+type Config struct {
+	HTTP        HTTPConfig        `mapstructure:"http"`
```
[the remark this answers is not in the document]
Comment (piavgh) [reply pkg/source/hashflow/client/config.go:9]: for example, `redisCache` => `redis_cache`

File: pkg/source/hashflow/client/memory_cache.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+		}
+
+		pairs = append(pairs, marketMakerPairs.([]hashflow.Pair)...)
+	}
+
+	return pairs, nil
+}
+
+func (c *memoryCacheClient) savePairsByMarketMakerToCache(marketMaker string, pairs []hashflow.Pair) error {
+	c.cache.SetWithTTL(marketMaker, pairs, 1, c.config.TTL.Duration)
```
Comment (piavgh) [inline pkg/source/hashflow/client/memory_cache.go:94]: this magic number `1` is hard to understand here, can you please add a comment to explain it, or better add a constant for it?

File: pkg/source/hashflow/reserves.go [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+	amountOut, _ := new(big.Float).Mul(
+		amountOutAfterDecimals,
+		bignumber.TenPowDecimals(pair.Decimals[1]),
+	).Int(nil)
+
+	return amountOut
+}
+
+func getAmountOut(amountIn *big.Float, priceLevels []PriceLevel) *big.Float {
```
Comment (piavgh) [inline pkg/source/hashflow/reserves.go:55]: this `getAmountOut` function name is a bit misleading.
Its meaning is something like `getMaxLiquidity`

**Review by hanguyenk — APPROVED** [review]

**Comment by lehainam-dev** [comment]
Close this PR since it is replaced by a new Hashflow integration.
Check out EX-1116 & EX-1117.

**Pull request closed without merging**